server/src/main/java/org/elasticsearch/action/admin/indices/resolve/ResolveClusterActionRequest.java

```diff
@@ -168,7 +168,10 @@ boolean localIndicesPresent(String[] indices) {
         for (String index : indices) {
             // ensure that `index` is a remote name and not a date math expression which includes ':' symbol
             // since date math expression after evaluation should not contain ':' symbol
-            String indexExpression = IndexNameExpressionResolver.resolveDateMathExpression(index);
+            // NOTE: index expressions can be prefixed with "-", which will not be parsed by resolveDateMathExpression,
+            String indexExpression = IndexNameExpressionResolver.resolveDateMathExpression(
+                index.charAt(0) == '-' ? index.substring(1) : index
+            );
```

Comment:
nit: can you clarify in the comment what the `-` means? I think it means to exclude it from the search, right?
Also: nit: the code comment ends with a comma, which could be removed